### ISSUE TITLE
directory listing link href's are not properly escaped

### ISSUE THREAD

**Issue by nestor-custodio** [joseluisq/static-web-server#149]
**Describe the bug**
When generating the directory listing HTML, href values for file/directory links are not being escaped, leading to either (a) *dead links in the directory listing*, or even worse (b) links in the directory listing that actually *point to a different file than the one shown*.

Additionally, names with a ':' are not handled properly by some browsers (`<a href="a:b">a:b</a>` is seen as a link to schema 'a' and path 'b', or possibly as a malformed URI of the "name:password@hostname" form), making them unusable. Any such files should instead be href'd with a './' prefix to help the browser resolve the name correctly.

**To Reproduce**
```shell
# start with clean workspace
tmp_dir="$( mktemp -d )"
pushd "$tmp_dir"

# create files
echo 'clicking this file in the directory listing will yield a 404' > 'unreachable-file?'
echo 'this is the file you get for "some-file", "some-file?", and "some-file?jkl" 😬' > 'some-file'
echo 'again, unreachable' > 'some-file?'
echo 'again, unreachable' > 'some-file?jkl'
echo "Chrome (and presumably other browsers) doesn't know what to do with the link for this file" > 'my:file'

# go!
static-web-server --root='./' --port='3000' --directory-listing='true'

# clean up
popd
rm -rf "$tmp_dir"
```

**Expected behavior**
File/directory names should be properly escaped so that non-URI-safe characters (like '?', ':', ' ', etc) are properly encoded and clicking the directory listing links yields the correct file.

**Environment and Specs**
 - static-web-server: v2.12.0
 - OS: Ubuntu Jammy (22.04.1)
 - Client: Chrome 105.0.5195.134

**Additional context**
Yes, I realize these are unorthodox filenames. They are still, however, valid in most filesystems and therefore should not break `static-web-server`, IMO. For whatever it's worth, I'm submitting the bug not because I'm poring through code and found something to nitpick about, but because I actually ran into the above problems in my usage.  🤷

**Comment by joseluisq** [comment]
Sure. Those URLs are simply not encoded therefore leading the browser to interpret them deliberately.
But in general, URL links that rely on directory/file names should be properly encoded.

I will also label this as a bug so a fix PR will come soon.

**Comment by joseluisq** [comment]
PR #150 fixes the issue. Thanks for reporting! :+1:

**Comment by joseluisq** [comment]
Released on [v2.13.0](https://github.com/joseluisq/static-web-server/releases/tag/v2.13.0)
